core/src/queue.rs

```diff
@@ -27,16 +27,16 @@ use crate::{prelude::*, tx::CheckSignatureCondition as _};
 /// Multiple producers, single consumer
 #[derive(Debug)]
 pub struct Queue {
-    /// The queue for transactions that passed signature check
+    /// The queue for transactions
     queue: ArrayQueue<HashOf<VersionedSignedTransaction>>,
-    /// The queue for transactions that didn't pass signature check and are waiting for additional signatures
-    ///
-    /// Second queue is needed to prevent situation when multisig transactions prevent ordinary transactions from being added into the queue
-    signature_buffer: ArrayQueue<HashOf<VersionedSignedTransaction>>,
-    /// [`VersionedAcceptedTransaction`]s addressed by `Hash`.
+    /// [`VersionedAcceptedTransaction`]s addressed by `Hash`
     txs: DashMap<HashOf<VersionedSignedTransaction>, VersionedAcceptedTransaction>,
+    /// Amount of transactions per user in the queue
+    txs_per_user: DashMap<AccountId, usize>,
     /// The maximum number of transactions in the queue
     max_txs: usize,
+    /// The maximum number of transactions in the queue per user. Used to apply throttling
+    max_txs_per_user: usize,
```

Comment:
Alternatively we can use `Option<NonZeroUsize>`.
And skip user transaction number check if `max_txs_per_user` is `None`

Reply:
I don't think this should ever be skipped. It's more appropriate the way you've done it

Reply:
I'd prefer that.